ui/packages/components/src/RunDetailsV3/LegacyRunsToggle.tsx

```diff
@@ -0,0 +1,27 @@
+import { Pill } from '../Pill';
+import { useLegacyTrace } from '../Shared/useLegacyTrace';
+import { Switch } from '../Switch';
+
+export const LegacyRunsToggle = ({ traceAIEnabled }: { traceAIEnabled: boolean }) => {
+  const {
+    enabled: legacyTraceEnabled,
+    ready: legacyTraceReady,
+    toggle: toggleLegacyTrace,
+  } = useLegacyTrace();
+  return (
+    traceAIEnabled &&
+    legacyTraceReady && (
+      <div className="flex flex-row items-center justify-end gap-2">
+        <Pill kind="info" appearance="solid" className="h-6">
+          Beta feature
+        </Pill>
+        <span className="text-sm">New runs view</span>
+        <Switch
+          checked={!legacyTraceEnabled}
+          className="data-[state=checked]:bg-primary-moderate cursor-pointer"
```

Comment:
Nitpick: I think we might not need the `data-[state=checked]:bg-primary-moderate` anymore, as it should be the default color already